demonstrations/tutorial_ibm_pennylane.py

```diff
@@ -0,0 +1,250 @@
+    qml.Hadamard(wires=2)
+    qml.Hadamard(wires=3)
+
+    qml.CNOT(wires=[3, 2])
+    qml.CNOT(wires=[2, 1])
+    qml.CNOT(wires=[1, 0])
+
+    qml.RZ(theta, wires=0)
+
+    qml.CNOT(wires=[1, 0])
+    qml.CNOT(wires=[2, 1])
+    qml.CNOT(wires=[3, 2])
+
+    # invert change of basis
+    qml.RX(-np.pi / 2, wires=0)
+    qml.Hadamard(wires=1)
+    qml.Hadamard(wires=2)
+    qml.Hadamard(wires=3)
+
+
+##############################################################################
+# Finally, we can run our example VQE algorithm, by using the vqe_runner function. It has many
+# options, which you can specify, such as the number of shots, the maximum number of iterations
+# and the initial values of the parameters.
+
+try:
+    job = vqe_runner(
+        program_id=program_id,
+        backend="ibmq_qasm_simulator",
+        hamiltonian=H,
+        ansatz=four_qubit_ansatz,
+        x0=[0.0],
+        shots=8000,
+        optimizer="SPSA",
+        optimizer_config={"maxiter": 30},
+        kwargs={"hub": "ibm-q", "group": "open", "project": "main"},
+    )
+
+except:
+    print("Error! Make sure you have a valid program ID!")
+
+##############################################################################
+# The results are saved in the job variable, in SciPy optimization format. You can also check the
+# results produced by any IBM device by logging in to your IBMQ account.
+
+##############################################################################
+# Benchmarking
+# ~~~~~~~~~~~~
+# One of the reasons why we even want to have access to these various devices and backends is so
+# that we can benchmark the capabilities of the algorithms that we develop. Some simulators are
+# particularly good with certain types of circuits, whereas other simulators are more general and
+# may provide resources for simulating noise which mimics the kind of errors that real quantum
+# hardware produces. Switching between your devices helps you learn more about your algorithm and
+# can potentially provide guidance on how to make it better. For example, we can compare the
+# performance of the default pennylane simulator to the qiskit aer_simulator by running the same
+# VQE algorithm on both. The difference between these two devices is that the aer_simulator uses a
+# finite number of shots to estimate the energy in each iteration, rather than performing an exact
+# calculation using the information hidden in the vector representation of the quantum state.
+
+dev1 = qml.device("default.qubit", wires=4)
+dev2 = qml.device("qiskit.aer", wires=4, backend="aer_simulator")
+
+
+@qml.qnode(dev1)
+def cost_fn_1(theta):
+    four_qubit_ansatz(theta)
+    return qml.expval(H)
+
+
+@qml.qnode(dev2)
+def cost_fn_2(theta):
+    four_qubit_ansatz(theta)
+    return qml.expval(H)
+
+
+stepsize = 0.4
+max_iterations = 40
+opt = qml.GradientDescentOptimizer(stepsize=stepsize)
+theta_1 = np.zeros(1, requires_grad=True)
+theta_2 = np.zeros(1, requires_grad=True)
```

Comment:
```suggestion
theta_1 = np.array(0., requires_grad=True)
theta_2 = np.array(0., requires_grad=True)
```
This should solve the current issue